### PR TITLE
Bolt table to web component

### PR DESCRIPTION
**Pull request by adamszalapski**
## Jira

http://vjira2:8080/browse/BDS-658

## Summary

Conversion bolt table component to web component

## Details

Add web component support to bolt table.

## How to test

Run this code locally and check if all bolt table examples are generated correctly with js and without.
Check web component examples on the bottom of bolt table component page.

### CODE REVIEW

File: docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  <bolt-code-snippet syntax="dark" lang="html" {{ copy ? 'copy-to-clipboard' : '' }}>{% spaceless %}
+      {{ code | replace({
+        '<': '&lt;',
+        '>': '&gt;',
+      }) | trim | raw }}
+    {% endspaceless %}</bolt-code-snippet>
+{% endmacro %}
+
+{% import _self as table_example %}
+
```
Comment (mikemai2awesome) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig]: @adamszalapski Nice work! I would recommend the following to make things a little more consistent.

1. For the first example, please replicate the table **With Headers and Footer** here: https://master.boltdesignsystem.com/pattern-lab/?p=components-table
2. For the second example, please replicate this table: https://master.boltdesignsystem.com/pattern-lab/?p=components-table-cell-attributes
3. For the third example, please replicate this table: https://master.boltdesignsystem.com/pattern-lab/?p=components-table-item-variations

By doing so, we can clearly show that there is no visual differences between using the twig and web component.

Comment (adamszalapski) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig]: Hi, @mikemai2awesome i update the docs, please check right now. We don't have `bolt-header` as a web component that i use `bolt-text` for creating a cell in the last table.

Comment (adamszalapski) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig]: This one: 
image

**Comment by adamszalapski** [comment]
I have noticed one thing elements that not have web component version and they are added to the table they don't have correct styling.
HEADER in a web-component version of the table:
image
HEADER in twig version of the table:
image

**Comment by sghoweri** [comment]
> I have noticed one thing elements that not have web component version and they are added to the table they don't have correct styling.
> HEADER in a web-component version of the table:
> image
> image
> HEADER in twig version of the table:
> image
> image

@adamszalapski that's because without using `<slot>s` (which, as we chatted about, while possible, will likely require quite a bit of work), every single nested component inside the table needs to bring it's own encapsulated styles with it in order to continue to work.

For example, `<bolt-image>`, `<bolt-list>`, `<bolt-ol>`, `<bolt-ul>`, and `<bolt-text>` should all be working as expected since all those are web components that have style encapsulation built in.

File: docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig (hunk 1 above)
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig]: @adamszalapski That is the equivalent of the small headline, you can build it with `bolt-text`:

```html
<bolt-text headline font-size="small" text-transform="uppercase" letter-spacing="wide">
  Build 2.0.1.0 — SR-49969
</bolt-text>
```

**Comment by mikemai2awesome** [comment]
@adamszalapski Could you update the copy and structure (title > description > visual > code) on the web component demo page as the following?

Screen Shot 2019-04-02 at 12 47 00 PM
Screen Shot 2019-04-02 at 12 47 09 PM
Screen Shot 2019-04-02 at 12 47 17 PM

File: docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig (hunk 1 above)
Comment (adamszalapski) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/table/40-table-with-web-component.twig]: 👍

**Comment by adamszalapski** [comment]
@mikemai2awesome copy and structure updated.

**Pull request merged**